app/controllers/api/submissions_controller.rb

```diff
@@ -8,14 +8,44 @@ class SubmissionsController < ApplicationController
     before_action :authenticate
 
     def create
-      result = { result: true, message: public_app_params['jwt'] }
-      render(json: result.to_json)
+      online_submission = OnlineApplicationBuilder.new(public_app_params).build
+      if online_submission.save
+        render(json: { result: true, message: online_submission.reference }.to_json)
+      else
+        render(json: { result: false, message: 'Could not save online_submission' }.to_json)
+      end
     end
 
     private
 
+    # rubocop:disable MethodLength
     def public_app_params
-      params.permit(:jwt)
+      params.require(:jwt).permit(
```

Comment:
This goes throughout the PR, but I was thinking about calling the parameter something more current, like 'online_application', or 'application', ... Just in case we don't get to do the jwt soon.